test/e2e/specs/add-embed-blocks.test.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Internal dependencies
+ */
+import '../support/bootstrap';
+import { newPost, newDesktopBrowserPage } from '../support/utils';
+
+describe( 'embedding media', () => {
+	beforeEach( async () => {
+		await newDesktopBrowserPage();
+		await newPost();
+		await page.click( '.edit-post-header [aria-label="Add block"]' );
+		await page.keyboard.type( 'embed' );
+		await page.keyboard.press( 'Tab' );
+		await page.keyboard.press( 'Tab' );
+		await page.keyboard.press( 'Enter' );
+	} );
+
+	afterEach( async () => {
+		//reset view to visual editor for next test
+		await page.click( '.edit-post-more-menu [aria-label="More"]' );
+		const visualEditorButton = ( await page.$x( '//button[contains(text(), \'Visual Editor\')]' ) )[ 0 ];
+		await visualEditorButton.click( 'button' );
+	} );
+
+	it( 'Should insert an embedded tweet', async () => {
+		//insert a twitter link
+		await page.keyboard.type( 'https://twitter.com/WordPress/status/980716623060455424' );
+		await page.keyboard.press( 'Tab' );
+		await page.keyboard.press( 'Enter' );
+
+		//wait for link to resolve and block to rerender
+		await page.waitFor( 2000 );
```

Comment:
Are we guaranteed that 2s is enough time for embed to occur? This could result in intermittent failures which are difficult to debug. Instead, could we wait for some condition to be satisfied to conclude that loading has finished? Maybe a [`load` event](https://developer.mozilla.org/en-US/docs/Web/Events/load) on the iframe? Not sure if Puppeteer has anything for this which would be as convenient as [`page.waitForSelector`](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagewaitfornavigationoptions).